File: backoffice/src/shared/domain/valueObjects/StringValueObject.py
"""
 *
 * Class 
 *
"""

class StringValueObject:

    """
     *
     * Attributes 
     *
    """

    __value : str

    """
     *
     * Methods 
     *
    """

    def __init__( self, value : str ) -> object:
        self.__value = value
    
    def getValue( self ) -> str:
        return self.__value

    def isEmpty( self ) -> bool:
        return self.__value == ''
    
    def equals( self, other : object ) -> bool:
        return self.__value == other.getValue()
    
    def isLongerThan( self, other : object ) -> bool:
        return len( self.__value ) > len( other.getValue() )
    
    def isLongerThanOrEqual( self, other : object ) -> bool:
        return len( self.__value ) >= len( other.getValue() )

File: preparation/src/shared/domain/valueObjects/StringValueObject.cs
/* 
 * 
 * Class
 *
*/

namespace preparation.src.shared.domain {

    public class StringValueObject {

        /* 
         * 
         * Attributes
         *
        */

        private string value;

        /* 
         * 
         * Methods
         *
        */

        public StringValueObject( string value ) {
            this.value = value;
        }

        public string getValue() {
            return value;
        }

        public bool isEmpty() {
            return value == string.Empty;
        }

        public bool equals( StringValueObject other ) {
            return value == other.getValue();
        }

        public bool isLongerThan( StringValueObject other ) {
            return value.Length > other.getValue().Length;
        }

        public bool isLongerThanOrEqual( StringValueObject other ) {
            return value.Length >= other.getValue().Length;
        }

    }

}